packages/core/src/database/utils.ts

```diff
@@ -0,0 +1,23 @@
+import { conditionalSql } from '@logto/shared';
+import { sql } from 'slonik';
+
+/**
+ * Options for searching for a string within a set of fields (case-insensitive).
+ *
+ * Note: `id` is excluded from the fields since it should be unique.
+ */
+export type SearchOptions<Keys extends string> = {
+  fields: ReadonlyArray<Exclude<Keys, 'id'>>;
+  keyword: string;
+};
+
+export const buildSearchSql = <SearchKeys extends string>(search?: SearchOptions<SearchKeys>) => {
```

Comment:
nit: add some comments indicating this method will build a `or` join SQL instead of `any`

Reply:
added to the to-do list